File: src/main/java/com/adjorno/billib/rest/db/Week.java
package com.adjorno.billib.rest.db;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;

@Entity
@Table(name = "WEEK")
public class Week {
    @Id
    @Column(name = "WEEK_ID")
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long mId;

    @Column(name = "DATE")
    private String mDate;

    public Week() {
    }

    public Week(Long id, String date) {
        mId = id;
        mDate = date;
    }

    @JsonIgnore
    public Long getId() {
        return mId;
    }

    public void setId(Long id) {
        mId = id;
    }

    public String getDate() {
        return mDate;
    }

    public void setDate(String date) {
        mDate = date;
    }

    @Override
    public String toString() {
        return mDate;
    }
}

File: src/main/java/com/adjorno/billib/rest/db/Week.kt
package com.adjorno.billib.rest.db

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "WEEK")
data class Week(
    @Id
    @Column(name = "WEEK_ID")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @Column(name = "DATE")
    var date: String? = null
) {
    override fun toString() = date.toString()
}
